examples/user_profiles/connector.py

```diff
@@ -0,0 +1,143 @@
+from fivetran_connector_sdk import Connector
+from fivetran_connector_sdk import Operations as op
+from fivetran_connector_sdk import Logging as log
+import requests as rq
+import pandas as pd
+
+
+def schema(configuration: dict):
+    return [
+        {
+            "table": "profile",
+            "primary_key": ["id"],
+            "columns": {
+                "id": "INT",
+                "gender": "STRING",
+                "fullName": "STRING",
+                "email": "STRING",
+                "age": "INT",
+                "date": "STRING",
+                "mobile": "STRING",
+                "nationality": "STRING"
+            }
+
+        },
+        {
+            "table": "location",
+            "primary_key": ["profileId"],
+            "columns": {
+                "profileId": "INT",
+                "street": "STRING",
+                "city": "STRING",
+                "state": "STRING",
+                "country": "STRING",
+                "postcode": "STRING"
+            }
+        },
+        {
+            "table": "Login",
+            "primary_key": ["profileId"],
+            "columns": {
+                "profileId": "INT",
+                "uuid": "STRING",
+                "username": "STRING",
+                "password": "STRING"
+            }
+        }
+    ]
+
+
+def update(configuration: dict, state: dict):
+    profile_cursor = state["profile_cursor"] if "profile_cursor" in state else 0
+
+    profile_df, location_df, login_df, cursor = get_data(profile_cursor)
+
+    # UPSERT all profile table data and checkpointing after every 5 records
+    for index, row in profile_df.iterrows():
+        yield op.upsert("profile", {col: row[col] for col in profile_df.columns})
+        if index % 5 == 0:
```

Comment:
This seems a bit contrived, why are you checkpointing like this?

Reply:
We should develop best practices for checkpointing - document them and then follow them in all our examples.

Reply:
I wrote this logic just to specify that they should perform a checkpoint after reading some number of records. Even though I can just checkpoint it at last, we shouldn't do that.

Reply:
We could tweak the comment:
# UPSERT all profile table data, checkpoint periodically to save progress. In this example every 5 records.